### PR TITLE
Feat/add rollup

### PR DESCRIPTION
**Pull request by traversng**
Original commit by @6Laq did not account for page status and passed template instead of the template id. 
- Updated to accept and pass status and id (template apiId) 

I discovered a bug with universal rendering and after talking with @Hendrixer , we noticed that typescript was not outputting in esm format.
- Added rollup to fix this

### CODE REVIEW

File: test/index.spec.ts [hunk 1]
```diff
@@ -21,88 +21,86 @@ describe('Tipe', () => {
...
           'Content-Type': 'application/json; charset=utf-8',
           Authorization: config.key
         },
-        cache: 'no-cache',
-        timeout: 5000
+        cache: 'no-cache'
       })
     })
   })
+
```
Comment (Hendrixer) [inline test/index.spec.ts]: remove this if not needed

File: src/index.ts [hunk 2]
```diff
@@ -17,27 +19,34 @@ export default class Client {
...
+  public getPageByParam = (pageConfig: IGetPageByParam, options?: ITipeClientOptions): Promise<{ [key: string]: any }> => {
+    return this.api(`POST`, `pageByParam`, pageConfig, options)
   }
 
-  public api: APIFetcher = (restMethod = 'GET', path, contentConfig, fetchConfig) => {
+  public getPagesByTemplate = (pageConfig: IGetPagesByTemplate, options?: ITipeClientOptions): Promise<{[key: string]: any}> => {
+    return this.api(`POST`, `pagesByTemplate`, pageConfig, options)
+  }
+
+  // public getPageByTipeId = (pageConfig: IGetPageByTipeIdOptions,  options?: ITipeClientOptions): Promise<{[key: string]: any}> => {
```
Comment (Hendrixer) [inline src/index.ts]: remove this if not needed

Comment (traversng) [reply src/index.ts]: @Hendrixer I thought we would be keeping this for page previews once we work on that feature?

Comment (Hendrixer) [reply src/index.ts]: oh yea, you right

**Pull request merged**